etna/models/nn/deepstate/deepstate.py

```diff
@@ -179,7 +181,7 @@ def forward(self, x: DeepStateBatch, *args, **kwargs):  # type: ignore
 
     def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: int) -> Iterator[dict]:
         """Make samples from segment DataFrame."""
-        values_real = df.drop(columns=["target", "segment", "timestamp"]).values
+        values_real = df.drop(columns=["target", "segment", "timestamp"]).select_dtypes(include=[np.number]).values
```

Comment:
This line leads to errors, because it starts ignoring `SegmentEncoderTransform()`.
If we keep previous version without `select_dtypes` it fails on `str` columns, which other models process without errors (they skip it).

Reply:
It is also related to #211

Reply:
> This line leads to errors, because it starts ignoring SegmentEncoderTransform().

What kinds of errors?

> If we keep previous version without select_dtypes it fails on str columns

Where do str come from?

Reply:
- If we use code in this PR column generated by `SegmentEncoderTransform` is ignored and the error occurs that `input_size` is wrong (because column is ignored)
- I put column with `str` type into input `TSDataset` for test `test_deepstate_make_samples` to check if model could work with it or not. Because other models are working fine with this kind of columns (they select only columns they can work with).

Reply:
You could pass categorical too